documentation.md

```diff
@@ -0,0 +1,117 @@
+
+Alternative Loss Functions Considered
+
+- NLLLoss: Negative Log-Likelihood Loss
+This function expands the binary cross-entropy loss. It requires an additional logSoftMax layer in the last layer of the network. As it outputs probablities, they must sum to 1.
+The function also has various reduction of output options such as the weighted mean of output or sum.
+
+- Sum of Log Loss: This function allows for the gradient to update with more incentive
+
+- Hinge: This function, nn.HingeEmbeddingLoss, is used more for measuring similarities. It tries to maximize the margin between decision boundary and data points. The main advantage is that the function penalizes incorrect predictions a lot but also correct ones that aren't confident (less). That is, confident correct predictions are not penalized at all.
+
+- L1 MAE: The mean absolute error
+
+### Structure of CNN Model
+
+Based on our Liturature Review, the following transfer learning architectures were considered:
+
+1. DenseNet
+2. Inception
+3. VGG
+4. ResNet
+
+The model variants that were considered are documented below, along with some notes thought to be relevant.
+
+- VGG16: It makes the improvement over AlexNet by replacing large kernel-sized filters (**11** and **5** in the first and second convolutional layer, respectively) with multiple 3×3 kernel-sized filters one after another.
+- DenseNet121: DenseNet121 proved to be the best performing model given the Lipohypertrophy data. To see this exploration, click [here](https://github.com/UBC-MDS/capstone-gdrl-lipo/blob/master/notebooks/model_decision_making.ipynb). 
+- ResNet50
+- InceptionV3
+
+#### Batch Normalization
+Ensure all models have batch normalization implemented. This will standardize the inputs of a network improving the overall efficiency by reducing the number of epochs required to train for a given model.
+
+#### Dropout Layer
+The implementation of dropout layers within the model's architecture to reduce the generalizable error were considered. This is due to the fact that a dropout layer with randomly with probably equal to the dropout rate, drop nodes from a network between layers. Given the Lipohypertrophy data, the dropout layers experiments with varying dropout rates have shown no success. To see this exploration, click [here](https://github.com/UBC-MDS/capstone-gdrl-lipo/blob/master/notebooks/densemodels-ax-dropout-layers.ipynb). 
+
+#### Recall
+To reduce the generalization error, consider varying the pos_weight argument in the loss function. Increasing pos_weight > 1 means there is a heavier penalization (loss) on positives in an attempt to reduce false negatives (a true positive where the model predicts is negative) and improve recall. This experumentation is also used to combat the class imbalance issue. To see this exploration, click [here](https://github.com/UBC-MDS/capstone-gdrl-lipo/blob/master/notebooks/pos-weight-exploration.ipynb).
```

Comment:
Suggestion: "To reduce the generalization error, we considered varying the pos_weight argument in the loss function. Increasing pos_weight > 1 means there is a heavier penalization (loss) on positives in an attempt to reduce false negatives (a true positive where the model predicts is negative) and improve recall. This experumentation is also used to combat the class imbalance issue. To see this exploration, click [here](https://github.com/UBC-MDS/capstone-gdrl-lipo/blob/master/notebooks/pos-weight-exploration.ipynb)".